### PR TITLE
feat: add interval sealing mode for development node (without relay)

### LINKED ISSUE #1204: Add Interval Sealing Option for No Relay Node
As a developer working with Frequency, I would like to have a local system that allows me to simulate the blocks forming every `n` seconds. (empty is optional)

## Notes
- Add to the manual sealing mode an option that will trigger a block every `n` seconds
- Example: `frequency --dev --sealing=manual --sealing-interval 12 --sealing-allow-empty-blocks`

## ToDo
- [x] Make it work
- [x] Documentation in the readme
- [x] Documentation in the instant seal docker readme
- [x] Parameter to allow or disallow empty blocks (default either way)

### PR DESCRIPTION
**Pull request by rlaferla**
# Goal
The goal of this PR is to add interval sealing mode for development node (no relay chain.)

Closes #1204

### Testing

`./target/debug/frequency --dev --sealing=interval --sealing-interval=30`

For regression testing:

`./target/debug/frequency --dev --sealing=instant `

`./target/debug/frequency --dev --sealing=manual `

```
USAGE:

      --sealing <SEALING>
          The sealing mode
          
          [default: instant]

          Possible values:
          - instant:
            Author a block immediately upon receiving a transaction into the transaction pool
          - manual:
            Author a block upon receiving an RPC command
          - interval:
            Author blocks at a regular interval specified in seconds

      --sealing-interval <SECONDS>
          The interval in seconds
          
          [default: 12]
   
```

### CODE REVIEW

File: node/service/Cargo.toml [hunk 1]
```diff
@@ -18,6 +18,7 @@ hex-literal = "0.3.4"
 jsonrpsee = { version = "0.16.2", features = ["server"] }
 log = "0.4.17"
 serde = { version = "1.0.151", features = ["derive"] }
+tokio = "1.28.1"
```
[the remark this answers is not in the document]
Comment (rlaferla) [reply node/service/Cargo.toml]: fixed

File: node/service/src/block_sealing.rs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	sealing_mode: SealingMode,
+	sealing_interval: u16,
+	sealing_allow_empty_blocks: bool,
+) -> Result<TaskManager, sc_service::error::Error> {
+	let extra;
+	if sealing_mode == SealingMode::Interval {
+		extra = format!(" ({}s interval)", sealing_interval);
+	} else {
+		extra = String::from("");
+	}
```
Comment (wilwade) [inline node/service/src/block_sealing.rs:28]: Rust-ism suggestion
```suggestion
	let extra = if sealing_mode == SealingMode::Interval {
		format!(" ({}s interval)", sealing_interval);
	} else {
		String::from("");
	}
```

File: scripts/init.sh [hunk 3]
```diff
@@ -264,4 +294,4 @@ upgrade-frequency-no-relay)
 
   ;;
 
-esac
+esac
```
Comment (wilwade) [inline scripts/init.sh:297]: Side note that I think most of us have new line at end of files enabled to keep this from happening.

https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

**Review by wilwade — COMMENTED** [review]
Two small notes. I haven't been able to run it yet, but should be able to later today.

File: node/service/src/block_sealing.rs (hunk 2 above)
Comment (rlaferla) [reply node/service/src/block_sealing.rs:28]: fixed (but your code was missing a semicolon)

File: scripts/init.sh (hunk 3 above)
Comment (rlaferla) [reply scripts/init.sh:297]: image

File: node/cli/src/cli.rs [hunk 4]
```diff
@@ -100,15 +100,6 @@ pub struct Cli {
 	#[cfg(feature = "frequency-no-relay")]
 	#[clap(long, help = "The interval in seconds", default_value = "120", value_name = "SECONDS")]
```
Comment (wilwade) [inline node/cli/src/cli.rs:101]: I think we want the default interval to be 12 seconds not 120 seconds

Comment (rlaferla) [reply node/cli/src/cli.rs:101]: done

**Review by wilwade — APPROVED** [review]
Nice.
- Works on my machine: ✅ 
- Works with different versions of the command: ✅ 
- Correctly only supports integers for seconds: ✅ 
- Good help messages: ✅

**Review by enddynayn — APPROVED** [review]
Nice! 👍

**Pull request merged**